source/Coop/Mod/Persistence/RPC/EventBroadcastingQueue.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Common;
+using JetBrains.Annotations;
+using NLog;
+using RailgunNet.Connection.Server;
+using Sync.Store;
+
+namespace Coop.Mod.Persistence.RPC
+{
+    /// <summary>
+    ///     Queue to broadcast events. Sends all events in order and makes sure, that all event
+    ///     arguments can be resolved on the clients. This includes waiting for any pending
+    ///     large object transfers.
+    /// </summary>
+    public class EventBroadcastingQueue : IUpdateable
```

Comment:
Can you add a max queue size and have the Timeout point to the server configuration

Reply:
Done. I added errors and an exception (DEBUG only) when the maximum queue size is reached. 

I was thinking it might be i good idea to automatically pause the game when the maximum queue size is reached. That would stop most other events from occuring and give the cause of the delay time to "catch up". That would be a somewhat elegant way to deal with low bandwith connections. What do you think?

Reply:
I'm fine with that.